include/seqan3/alignment/data_structure/scoring_scheme_base.hpp

```diff
@@ -0,0 +1,279 @@
+    constexpr scoring_scheme_base(scoring_scheme_base &&) noexcept = default;
+    constexpr scoring_scheme_base & operator=(scoring_scheme_base const &) noexcept = default;
+    constexpr scoring_scheme_base & operator=(scoring_scheme_base &&) noexcept = default;
+    ~scoring_scheme_base() noexcept = default;
+
+    //!\brief The default constructor (delegates to set_hamming_distance()).
+    constexpr scoring_scheme_base() noexcept
+    {
+        set_hamming_distance();
+    }
+
+    /*!\brief Constructor for the simple scheme (delegates to set_simple_scheme()).
+     * \copydetails set_simple_scheme()
+     */
+    template <arithmetic_concept score_arg_t>
+    constexpr scoring_scheme_base(match_score<score_arg_t> const ms, mismatch_score<score_arg_t> const mms)
+    {
+        set_simple_scheme(ms, mms);
+    }
+
+    /*!\brief Constructor for a custom scheme (delegates to set_custom_matrix()).
+     * \copydetails set_custom_matrix()
+     */
+    constexpr scoring_scheme_base(matrix_t const & _matrix) noexcept
+    {
+        set_custom_matrix(_matrix);
+    }
+    //!\}
+
+public:
+    /*!\name Scheme selection
+     * \{
+     */
+    //!\brief Set the hamming scheme, a variant of the simple scheme where match is scored `0` and mismatch `-1`.
+    constexpr void set_hamming_distance() noexcept
+    {
+        set_simple_scheme(match_score<score_t>{0}, mismatch_score<score_t>{-1});
+    }
+
+    /*!\brief Set the simple scheme (everything is either match or mismatch).
+     * \tparam score_arg_t The underlying type of the arguments.
+     * \param[in] ms  Matches shall be given this value (of type seqan3::match_score).
+     * \param[in] mms Mismatches shall be given this value (of type seqan3::mismatch_score).
+     * \throws std::invalid_argument Thrown if you pass a value that is to large/low to be represented by `score_t`.
+     */
+    template <arithmetic_concept score_arg_t>
+    constexpr void set_simple_scheme(match_score<score_arg_t> const ms, mismatch_score<score_arg_t> const mms)
+    {
+        std::conditional_t<integral_concept<score_t>, int64_t, double> i_ms = static_cast<score_arg_t>(ms);
+        std::conditional_t<integral_concept<score_t>, int64_t, double> i_mms = static_cast<score_arg_t>(mms);
+        if ((i_ms  < std::numeric_limits<score_t>::lowest() || i_ms  > std::numeric_limits<score_t>::max()) ||
+            (i_mms < std::numeric_limits<score_t>::lowest() || i_mms > std::numeric_limits<score_t>::max()))
+        {
+            throw std::invalid_argument{"You passed a score value to set_simple_scheme that is out of range of the "
+                                        "scoring scheme's underlying type. Define your scoring scheme with a larger "
+                                        "template parameter; or construct match_score explicitly with a template "
+                                        "parameter to silence this warning."};
```

Comment:
Mhmm, it is not just a warning, but an exception causing a hard fail if it is not catched. So I would remove the last part of the last sentence.

Reply:
I have changed the text so that it doesn't suggest a warning, but I think it still doesn't hurt to tell users how to circumvent this.